src/middlewared/middlewared/plugins/truenas_connect/hostname.py

```diff
@@ -50,20 +51,39 @@ async def sync_interface_ips(self):
         if response['error']:
             logger.error('Failed to update IPs with TrueNAS Connect: %s', response['error'])
 
+    async def handle_update_ips(self, event_type, args):
+        """
+        Handle IP address changes for TrueNAS Connect.
+        This method is called when an IP address change event occurs.
+        """
+        tnc_config = await self.middleware.call('tn_connect.config')
+        internal_interfaces = tuple(await self.middleware.call('interface.internal_interfaces'))
+
+        if args['fields']['iface'] is None or tnc_config['status'] not in CONFIGURED_TNC_STATES or (
+            tnc_config['use_all_interfaces'] is False and args['fields']['iface'] not in tnc_config['interfaces']
+        ) or args['fields']['iface'].startswith(internal_interfaces):
```

Comment:
This is kind of difficult to parse. Can we maybe have something like:

```
if <condition 1>:
    # comment explaining condition
    return    

if <condition 2>:
    # comment explaining condition
    return
```
Broken in up some reasonable number of chunks